hoss/spatial.py

```diff
@@ -187,6 +209,71 @@ def get_projected_x_y_index_ranges(
     return x_y_index_ranges
 
 
+def get_x_y_index_ranges_from_coordinates(
+    non_spatial_variable: str,
+    varinfo: VarInfoFromDmr,
+    prefetch_coordinate_datasets: Dataset,
+    coordinates: Set[str],
+    index_ranges: IndexRanges,
+    bounding_box: BBox = None,
+    shape_file_path: str = None,
+) -> IndexRanges:
+    """This function returns a dictionary containing the minimum and maximum
+    index ranges for a pair of lat/lon coordinates, e.g.:
+
+    index_ranges = {'/x': (20, 42), '/y': (31, 53)}
+
+    This method is called when the CF standards are not followed
+    in the source granule and only coordinate datasets are provided.
+    The coordinate datasets along with the crs is used to calculate
+    the x-y projected dimension scales. The dimensions of the input,
+    non-spatial variable are checked for associated coordinates. If
+    these are present, and they have not already been added to the
+    `index_ranges` cache, the extents of the input spatial subset
+    are determined in these projected coordinates. The minimum and
+    maximum values are then derived from these projected coordinate
+    points.
+
+    """
+    crs = get_variable_crs(non_spatial_variable, varinfo)
+
+    projected_x = 'projected_x'
+    projected_y = 'projected_y'
+    override_dimension_datasets = update_dimension_variables(
+        prefetch_coordinate_datasets, coordinates, varinfo, crs
+    )
+
+    if (
+        projected_x is not None
+        and projected_y is not None
```

Comment:
These conditions doesn't make sense to me. Isn't it just checking the string variables you are assigning a few lines earlier have been assigned. How could they be `None`?

(Sorry if I'm missing something obvious, here)

Reply:
right... - when I split it up from the combined method, should have removed that check

Reply:
updated - https://github.com/nasa/harmony-opendap-subsetter/pull/15/commits/802fe0ec04b853a7a7abb4718e2a05b0e8e6ed3c